sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -400,6 +415,14 @@ def _intilialize_root(self, gradients, hessians, hessians_are_constant):
         self.root.histograms = self.histogram_builder.compute_histograms_brute(
             self.root.sample_indices
         )
+
+        if self.interaction_cst is not None:
+            self.root.interaction_cst_indices = list(range(len(self.interaction_cst)))
```

Comment:
From the way we use `interaction_cst_indices`, I think we can get away with not creating the list? Something like this:

```suggestion
            self.root.interaction_cst_indices = range(len(self.interaction_cst))
```

I have no strong opinion on this.

Reply:
Generators, good idea, might help in a follow-up for several things.